File: csharp/ScheduleTime.Domain/Factories/BeautySalonFactory.cs
using ScheduleTime.Domain.Entities;
using ScheduleTime.Domain.Factories.Interfaces;

namespace ScheduleTime.Domain.Factories
{
    public class BeautySalonFactory : IScheduleFactory
    {
        public Schedule CreateSchedule(DateTime date, Customer customer)
        {
            var beautySalon = new BeautySalon();
            return new Schedule(date, customer, null, beautySalon);
        }
    }
}

File: java/schedule-time-event/src/main/java/com/gearguru/schedule_time_event/Domain/Factories/BeautySalonFactory.java
package com.gearguru.schedule_time_event.Domain.Factories;

import com.gearguru.schedule_time_event.Domain.Entities.BeautySalon;
import com.gearguru.schedule_time_event.Domain.Entities.Schedule;
import com.gearguru.schedule_time_event.Domain.Interfaces.IScheduleFactory;

public class BeautySalonFactory implements IScheduleFactory {
    @Override
    public Schedule scheduleTime() {
        return new BeautySalon();
    }
}
